datafusion/src/execution/context.rs

```diff
@@ -311,7 +311,7 @@ impl ExecutionContext {
         // create a query planner
         let state = self.state.lock().clone();
         let query_planner = SqlToRel::new(&state);
-        query_planner.statement_to_plan(&statements[0])
+        query_planner.statement_to_plan(statements.pop().unwrap())
```

Comment:
how about combining the length check and pop logic into a single block to make it more readable? basically something like below:

```rust
let stmt = statements.pop().ok_or(Err(DataFusionError::NotImplemented(
    "The context currently only supports a single SQL statement".to_string()
)))?;
```

Reply:
great